### PR TITLE
Add MapType enum to RPG::TreeMap

### PR DESCRIPTION
**Pull request by Ghabry**
-

### CODE REVIEW

File: generator/csv/enums.csv [hunk 1]
```diff
@@ -247,6 +247,8 @@ Map,GeneratorMode,maze_passage,2
 Map,GeneratorMode,open_room,3
 Map,GeneratorTiles,one,0
 Map,GeneratorTiles,two,1
+TreeMap,MapType,map,1
```
Comment (carstene1ns) [inline generator/csv/enums.csv]: Maybe we should also add `0` for completeness sake.

Comment (Ghabry) [inline generator/csv/enums.csv]: Makes sense because 0 is the root-type

**Comment by carstene1ns** [comment]
proposed addition:

```patch
diff --git a/generator/csv/enums.csv b/generator/csv/enums.csv
index 64e2df7..98cd21c 100644
--- a/generator/csv/enums.csv
+++ b/generator/csv/enums.csv
@@ -247,6 +247,7 @@ Map,GeneratorMode,maze_passage,2
 Map,GeneratorMode,open_room,3
 Map,GeneratorTiles,one,0
 Map,GeneratorTiles,two,1
+TreeMap,MapType,root,0
 TreeMap,MapType,map,1
 TreeMap,MapType,area,2
 MoveCommand,Code,move_up,0
diff --git a/generator/csv/fields.csv b/generator/csv/fields.csv
index d0ede80..20ade92 100644
--- a/generator/csv/fields.csv
+++ b/generator/csv/fields.csv
@@ -600,7 +600,7 @@ Encounter,troop_id,f,Ref<Troop>,0x01,0,Integer
 MapInfo,name,f,String,0x01,'',String. Note: Map ID 0 used to be game title but it should be ignored (TreeCtrl dummy editor dumped data); always use RPG_RT.ini GameTitle instead
 MapInfo,parent_map,f,Ref<Map>,0x02,0,Integer. Used to inherit parent map properties
 MapInfo,indentation,f,Integer,0x03,0,Integer. Dummy editor dumped data. Branch indentation level in TreeCtrl
-MapInfo,type,f,Integer,0x04,-1,Integer. 0=lmt root (can be ignored); 1=map; 2=area
+MapInfo,type,f,Enum<TreeMap_MapType>,0x04,-1,Integer
 MapInfo,scrollbar_x,f,Integer,0x05,0,Integer. Editor only
 MapInfo,scrollbar_y,f,Integer,0x06,0,Integer. Editor only
 MapInfo,expanded_node,f,Boolean,0x07,False,Flag. Editor only
diff --git a/src/generated/lmt_chunks.h b/src/generated/lmt_chunks.h
index e2b930f..0ea0f47 100644
--- a/src/generated/lmt_chunks.h
+++ b/src/generated/lmt_chunks.h
@@ -30,7 +30,7 @@ namespace LMT_Reader {
 			parent_map		= 0x02,
 			/** Integer. Dummy editor dumped data. Branch indentation level in TreeCtrl */
 			indentation		= 0x03,
-			/** Integer. 0=lmt root (can be ignored); 1=map; 2=area */
+			/** Integer */
 			type			= 0x04,
 			/** Integer. Editor only */
 			scrollbar_x		= 0x05,
diff --git a/src/generated/rpg_treemap.h b/src/generated/rpg_treemap.h
index 892e8d0..89323c7 100644
--- a/src/generated/rpg_treemap.h
+++ b/src/generated/rpg_treemap.h
@@ -24,6 +24,7 @@ namespace RPG {
 	class TreeMap {
 	public:
 		enum MapType {
+			MapType_root = 0,
 			MapType_map = 1,
 			MapType_area = 2
 		};
```

**Comment by Ghabry** [comment]
I don't really like that the comment is always just something like "Integer" but this is a general problem.

**Comment by Ghabry** [comment]
Hm I wanted to create a "created by carstene1ns and commited by Gab" commit but no idea how o.O

**Pull request merged**